app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -822,21 +822,16 @@ def prepare_vm_disk_attachment(disk_spec, storage_spec)
     # @param vm_service [OvirtSDK4::VmsService] The service that manages the virtual machine.
     # @param disk_specs [Array<Hash>] The specifications of the disks to remove.
     #
+    # TODO: Note that disk_spec['disk_name'] is actually the disks id not its name.
+    # we should change that because it is very confusing.
     def remove_vm_disks(vm_service, disk_specs)
       attachments_service = vm_service.disk_attachments_service
-
-      disk_attachment_by_disk_name_hash = attachments_service.list.collect do |attachment_service_obj|
-        attachment_service = attachments_service.attachment_service(attachment_service_obj.id)
-        [vm_service.connection.follow_link(attachment_service_obj.disk).name, attachment_service]
-      end.to_h
-
       disk_specs.each do |disk_spec|
         disk_spec = disk_spec.with_indifferent_access
-        attachment_service = disk_attachment_by_disk_name_hash[disk_spec['disk_name']]
-        unless attachment_service
-          raise "no disk with the name #{disk_spec['disk_name']} is attached to the vm: #{vm_service.get.name}"
-        end
+        attachment_service = attachments_service.attachment_service(disk_spec['disk_name'])
         attachment_service.remove(:detach_only => !disk_spec['delete_backing'])
+      rescue OvirtSDK4::NotFoundError
```

Comment:
@borod108 the ci fails becuse you've placed the rescue badly either put it to `begin rescue end` or if it supposed to be a whole method rescue, put it out of the block

so
```
def xy

array.each do |a|
end

rescue Class
end
```

the rescue can;t be nested to the `each` block

Reply:
Thank you!

Reply:
wow this is actually cool, it did work for another version of ruby, and actually did work in practice!